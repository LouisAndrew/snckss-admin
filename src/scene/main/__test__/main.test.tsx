import React, { Suspense } from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { FirebaseAppProvider } from 'reactfire'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Main from '..'
import config from '../../../lib/firebase-config'

describe('Main component', () => {
        afterEach(cleanup)
        const el: React.ReactElement = (
                <FirebaseAppProvider firebaseConfig={config}>
                        <Suspense fallback={<></>}>
                                <Main />
                        </Suspense>
                </FirebaseAppProvider>
        )
        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
