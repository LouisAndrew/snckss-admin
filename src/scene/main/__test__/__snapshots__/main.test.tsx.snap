// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component matches snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-3 col-lg-3 col-xl-3"
    >
      <ul
        className="tabs nav flex-column nav-tabs"
      >
        <li
          className="product nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
          >
            Create Products
             
          </a>
        </li>
        <li
          className="orders nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
          >
            Show Orders
             
          </a>
        </li>
      </ul>
    </div>
    <div
      className="col-md-9 col-lg-9 col-xl-9"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-4 col-xl-4"
            m="4"
          >
            <h3>
              Select what you want to create
            </h3>
            <ul
              className="nav flex-column nav-tabs"
            >
              <li
                className="categories nav-item"
              >
                <a
                  className="nav-link"
                  onClick={[Function]}
                >
                  Categories
                </a>
              </li>
              <li
                className="brands nav-item"
              >
                <a
                  className="nav-link"
                  onClick={[Function]}
                >
                  Brands
                </a>
              </li>
              <li
                className="products nav-item"
              >
                <a
                  className="nav-link"
                  onClick={[Function]}
                >
                  Products
                </a>
              </li>
            </ul>
          </div>
          <div
            className="col-lg-8 col-xl-8"
            m="8"
          >
            <div
              className="brands"
            >
              <div
                className="back icon"
                onClick={[Function]}
              >
                <svg
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M20 11v2H8l5.5 5.5l-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5L8 11h12z\\" fill=\\"currentColor\\"/>",
                    }
                  }
                  focusable={false}
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "transform": "rotate(360deg)",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </div>
              <h1>
                Brands
              </h1>
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="name"
                >
                  Brand Name
                </label>
                <input
                  className="form-control"
                  id="name"
                  onChange={[Function]}
                  placeholder="Add brand name here!"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="select"
                >
                  Select brand's category
                </label>
                <select
                  className="form-control"
                  id="select"
                  name="select"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value="default"
                  >
                    Select one
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="select"
                >
                  Select prodcuts
                </label>
                <select
                  className="form-control"
                  id="select"
                  name="select"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value="default"
                  >
                    Select multiple
                  </option>
                </select>
              </div>
              <button
                aria-label={null}
                className="submit-btn btn btn-primary btn-block"
                onClick={[Function]}
                type="button"
              >
                Post
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
